tests/examples/form_data.rs

```diff
@@ -0,0 +1,34 @@
+use hyper::client::Client;
+use hyper::header::ContentType;
+use hyper::status::StatusCode;
+use util::{read_body_to_string, read_url, run_example};
+
+#[test]
+fn form_data() {
+    run_example("form_data", |port| {
+        // Display form
+        let url = format!("http://localhost:{}/", port);
+        let s = read_url(&url);
+        assert!(s.contains(r#"<form>"#), "response didn't have a form");
+
+        // POST success
+        let url = format!("http://localhost:{}/confirmation", port);
+        let f = |body| {
+            let ref mut res = Client::new()
+                .post(&url)
+                .header(ContentType::form_url_encoded())
+                .body(body)
+                .send()
+                .unwrap();
+            read_body_to_string(res)
+        };
+        let s = f(r#"firstname=John&lastname=Doe&phone=911&email=john@doe.com"#);
+        assert!(s.contains(r#"John Doe 911 john@doe.com"#), "response didn't have an expected data");
+        let s = f(r#""#);
+        assert!(s.contains(r#"First name? Last name? Phone? Email?"#), "response didn't have an expected data");
```

Comment:
If you have time, it would be good to split this test into several to make intention clearer and for easier debugging in the failure case. (these assertions kind of get lost in the flow of things, especially when the test finishes with testing a failure case)